### ISSUE TITLE
The 2 dll's

### ISSUE THREAD

**Issue by jphellemons** [jishi/Jishi.Intel.SonosUPnP#3]
Is the source of the 2 referenced dll's available? Or is there a nuget?

**Comment by jishi** [comment]
Isn't that the ones that are in this folder? 

https://github.com/jishi/Jishi.Intel.SonosUPnP/tree/master/lib

**Comment by jishi** [comment]
This library had a lot of short-comings, to be honest. I re-wrote most of it without the intel upnp libraries, which I used in this project: https://github.com/jishi/Jishi.StreamToSonos

If you are thinking about using something and extending it for own use, I suggest looking at that as well. 

Come to think of it, the biggest issue with this library was that the Sonos system would subscribe all subscription for a certain IP if it found a dead endpoint, not just a specific subscription. This caused all kind of side-effects, however I think Sonos fixed that some time ago. Can't say for sure though, since all my other projects are using a different approach (making this less of a problem).

**Comment by jphellemons** [comment]
Will take a look at it. But:

> This is a very experimental program! Do not fork.

I was thinking about using Open.Nat nuget, but that is not usable for UWP projects. Open.Nat.Core is .net standard 2, but standard 2 and uwp is still in preview (visual studio 15.4 preview 4)

I do not have the diskspace to install 15.4 prev 4 next to my 15.3.5 (production) so I will take a look at your experimental program and perhaps even fork (sorry!)

Thank you for your update! 👍

**Comment by jishi** [comment]
Well, that warning was more for the actual software, not the Sonos interfacing stuff :) 

I mean, the NAT part of UPnP is very different from the AV-stuff that Sonos uses, so not sure if those would be applicable, to be honest.

**Comment by jphellemons** [comment]
What would you suggest for me? I want an app on my mobile (Lumia 930) running Win10 creator update. So I was building an UWP app, but cannot reference any of the old projects it seems. Also the Open.Nat nugets do not work without the Visual Studio Preview for UWP projects.
I have also seen SonosSharp on GitHub, but that is also rather old.
All I want is to be able to press a big play button on my phone to restart the internet radio on my Sonos haha. I think it can be done today with Windows.Devices.Enumeration.DeviceInformation.CreateWatcher and subscribe to the upnp device

**Comment by jishi** [comment]
Win10 on the phone allows pure socket nowadays, correct? That was a problem before, from what I've heard, making UPnP very hard.

The library that is used for StreamToSonos only uses one external dependency, which is MonoHttpListener. Most of the other stuff is pure .NET 4.5, but not sure how much of that is in .NET core nowadays (don't do .NET development anymore). 

https://github.com/jishi/Jishi.StreamToSonos/tree/master/Jishi.SonosUPnP

This library already implements SetAVTransportURI (which you use to select a radio), and Play, so it should suffice given that you can get it to run. I know it's hard to use a library that has 0 documentation. 

Basically, you instantiate a `SonosDiscovery()`, you waith for the event `TopologyChanged` happens, then `.Zones` and `.Players` are populated. Given a Player, you can invoke `player.Play()` (which is async), and also `player.SetAVTransportUri()` with the uri pointing to the radio station you want. I don't think you need any metadata for radio stations. 

However, given that you need to "discover" players or this to function properly, it is ill suited for a one-off trigger. If you can live with hard-coding an IP address, you can get away by just invoking two very specific http requests with SOAP payloads, simplifying it quite a bit.

**Comment by jishi** [comment]
As for a more extreme solution, you could spin up a rpi with my simplified http api (there is a pre-compiled SD-card image already), and benefit from the "presets" functionality, which basically makes it possible with a single http request to create the desired grouping, volume and source of your Sonos system (unless you only have a single speaker, you'd probably want that eventually anyway). Would also make a phone button super simple to implement. 

Just as an option. Of course, maybe not the challenge you are looking for :)

**Comment by jphellemons** [comment]
Thank you (again) for the info! I tried to reference your `SonosUPnP` but somehow my VisualStudio says it cannot add a reference to it. There is no info in the output window in VS.... probably an issue on my side.
<img width="964" alt="knipsel" src="https://user-images.githubusercontent.com/1119134/31129809-f3e7470a-a855-11e7-95b5-59f0ccfd899a.PNG">

**Comment by jphellemons** [comment]
Removed some references in the lib and added the nuget libs. Does compile fine.
<img width="502" alt="knipsel" src="https://user-images.githubusercontent.com/1119134/31129949-57043442-a856-11e7-806f-41e809129fd4.PNG">

**Comment by jishi** [comment]
Hm... verify that the Nowin is actually used. I don't think it is.
